pkg/debug/start_debug.go

```diff
@@ -0,0 +1,169 @@
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deployment.Spec,
+	}
+
+	debugDeployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Create(ctx.TODO(), debug_deployment, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debug_deployment, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := setDeploymentScale(context, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func setDeploymentScale(context *k8sutil.Context, clusterNamespace, deploymentName string, scaleCount int) error {
+	scale, err := context.Clientset.AppsV1().Deployments(clusterNamespace).GetScale(ctx.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get scale of deployment %s. %v\n", deploymentName, err)
+	}
+
+	scale.Spec.Replicas = int32(scaleCount)
+	_, err = context.Clientset.AppsV1().Deployments(clusterNamespace).UpdateScale(ctx.TODO(), deploymentName, scale, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update scale of deployment %s. %v\n", deploymentName, err)
+	}
+	return nil
+}
+
+func verifyDeploymentExists(context *k8sutil.Context, clusterNamespace, deploymentName string) (*appsv1.Deployment, error) {
+	deployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Get(ctx.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
+}
+
+func updateDeployment(context *k8sutil.Context, clusterNamespace string, deploymentName *appsv1.Deployment) error {
+	_, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Update(ctx.TODO(), deploymentName, v1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func podRunningWithLabel(context *k8sutil.Context, clusterNamespace string, deploymentSpec appsv1.DeploymentSpec) (string, error) {
+
+	pods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), v1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Labels["ceph_daemon_type"] == deploymentSpec.Template.Labels["ceph_daemon_type"] &&
+			pod.Labels["ceph_daemon_id"] == deploymentSpec.Template.Labels["ceph_daemon_id"] {
+			return pod.Name, nil
+		}
+	}
+	return "", fmt.Errorf("No pod with labels matching %s:%s", deploymentSpec.Template.Labels, deploymentSpec.Template.Labels)
+}
+
+func waitForPodDeletion(context *k8sutil.Context, clusterNamespace, podName string) error {
+	for i := 0; i < 60; i++ {
+		_, err := context.Clientset.CoreV1().Pods(clusterNamespace).Get(ctx.TODO(), podName, v1.GetOptions{})
+		if kerrors.IsNotFound(err) {
+			return nil
+		}
+
+		err = context.Clientset.CoreV1().Pods(clusterNamespace).Delete(ctx.TODO(), podName, v1.DeleteOptions{})
```

Comment:
You could move this before the `for` loop, then your loop is just to get and sleep

Reply:
Actually, do you even need to delete the pod? I think you don't even want to delete the pod since the deployment scale-down should automatically delete it. So this method is just waiting for the pod to go away.

Reply:
right, update